draft-ietf-tls-esni.md

```diff
@@ -765,63 +759,50 @@ error code.
 
 ### Handling HelloRetryRequest {#client-hrr}
 
-As required in {{real-ech}}, clients offering ECH MUST ensure that all
-extensions or parameters that might change in response to receiving a
-HelloRetryRequest have the same values in ClientHelloInner and
-ClientHelloOuter. That is, if a HelloRetryRequest causes a parameter to be
-changed, the same change is applied to both ClientHelloInner and
-ClientHelloOuter. Applicable parameters include:
-
-1. TLS 1.3 {{!RFC8446}} ciphersuites in the ClientHello.cipher_suites list.
-1. The "key_share" and "supported_groups" extensions {{RFC8446}}. (These
-extensions may be copied from ClientHelloOuter into ClientHelloInner as
-described in {{real-ech}}.)
-1. Versions in the "supported_versions" extension, excluding TLS 1.2 and
-earlier. Note the ClientHelloOuter MAY include these older versions, while the
-ClientHelloInner MUST omit them.
-
-Future extensions that might change across first and second ClientHello messages
-in response to a HelloRetryRequest MUST have the same value.
-
-If the server sends a HelloRetryRequest in response to the ClientHello, the
-client sends a second updated ClientHello per the rules in {{RFC8446}}.
-However, at this point, the client does not know whether the server processed
-ClientHelloOuter or ClientHelloInner, and MUST regenerate both values to be
-acceptable. Note: if ClientHelloOuter and ClientHelloInner use different groups
-for their key shares or differ in some other way, then the HelloRetryRequest
-may actually be invalid for one or the other ClientHello, in which case a fresh
-ClientHello MUST be generated, ignoring the instructions in HelloRetryRequest.
-Otherwise, the usual rules for HelloRetryRequest processing apply.
-
-The client encodes the second ClientHelloInner as in {{encoding-inner}}, using
-the second ClientHelloOuter for any referenced extensions. It then encrypts
-the new EncodedClientHelloInner value as a second message with the previous
-HPKE context:
+When the server sends a HelloRetryRequest, the client determines if ECH was
+accepted by checking the message for an "encrypted_client_hello" extension with
+an 8-byte payload equal to `hrr_accept_confirmation` as defined in
+{{backend-server}}. If found, the client presumes acceptance and handles the
+HelloRetryRequest using ClientHelloInner. Otherwise, it presumes rejection and
+handles the HelloRetryRequest using ClientHelloOuter. Note that the
+client-facing server does not send "encrypted_client_hello" in case of
+rejection.
```

Comment:
```suggestion
handles the HelloRetryRequest using ClientHelloOuter. 
```

I thought the intent was to grease this in the case of rejection to ensure the extension is always present -- am I misremembering? @davidben, do you remember? (This would require us to add new text to the client-facing server section on what to do when it rejects but sends HRR, which we currently don't have.)

Reply:
I don't recall making a decision about this. I'd like to suggest we leave this change out of the current PR and leave an OPEN ISSUE for deciding whether to grease the HRR acceptance signal.

Reply:
That works for me, then. Let's prep that PR to build on top of this.

Reply:
Done.